Sources/AnyLanguageModel/Models/CoreMLLanguageModel.swift

```diff
@@ -8,16 +8,28 @@ import Tokenizers
 public struct CoreMLLanguageModel: AnyLanguageModel.LanguageModel {
     private let model: Models.LanguageModel
     private let tokenizer: Tokenizer
-
-    public init(url: URL, computeUnits: MLComputeUnits = .all) async throws {
-        // Load the CoreML model from the specified path and compile it, if needed
-        let compiledURL = try compileModel(at: url)
+    private let chatTemplateHandler: (@Sendable (Instructions?, Prompt) -> [Message])?
+    private let toolsHandler: (@Sendable ([any Tool]) -> [ToolSpec])?
+
+    public init(
+        url: URL,
+        computeUnits: MLComputeUnits = .all,
+        chatTemplateHandler: (@Sendable (Instructions?, Prompt) -> [Message])? = nil,
+        toolsHandler: (@Sendable ([any Tool]) -> [ToolSpec])? = nil
```

Comment:
@pcuenca How universal are chat template and tool calling syntax across models? My sense is that most models have overlapping conventions, but I don't know to what extent those can be generalized. Is this something we can determine from the tokenizer or other model config?